components/camera/transformpipeline/mods.go

```diff
@@ -187,16 +188,44 @@ func newCropTransform(
 	if conf.YMin >= conf.YMax {
 		return nil, camera.UnspecifiedStream, errors.New("cannot crop image to 0 height (y_min is >= y_max)")
 	}
-	cropRect := image.Rect(conf.XMin, conf.YMin, conf.XMax, conf.YMax)
+	cropRect := image.Rectangle{}
+	cropRel := []float64{}
+	if conf.XMax < 1. || conf.YMax < 1. {
+		if conf.XMin <= 1. && conf.YMin <= 1. && conf.XMax <= 1. && conf.YMax <= 1. {
+			cropRel = []float64{conf.XMin, conf.YMin, conf.XMax, conf.YMax}
+		} else {
+			return nil, camera.UnspecifiedStream, errors.New("if using relative bounds between 0 and 1 for cropping, all parameter attributes must be between 0 and 1")
```

Comment:
If XMin is less than 1 but XMax and YMax are not, we're going to treat them all like ints. Is that the right behavior? I kinda want the logic to be:
```
if all the values are less than 1, use relative cropping
else if any of the values are less than 1, it's an error
else use absolute cropping
```

Reply:
someone might want xMin or yMin to be 0, which are less than 1.0

Reply:
Didn't quite use this logic, because (0,0) (20, 30) is a valid cropping but would have failed condition 2, but I think I got a valid flow